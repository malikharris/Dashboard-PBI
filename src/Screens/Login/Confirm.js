/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { Auth } from 'aws-amplify';
import image from '../../assets/images/Login_Image.svg';
import logo from '../../assets/images/CORE-logo.svg';

import FormField from '../../comps/FormField/FormField';
import CustomButton from '../../comps/CustomButton/CustomButton';
import './Login.css';

export default class Confirm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      confirmationCode: '',
      confirmed: false,
    };
  }

  onChangeCode = e => {
    this.setState({ confirmationCode: e.target.value });
  };

  onConfirm = () => {
    const { confirmationCode } = this.state;
    const {
      location: { search },
    } = this.props;
    let username = '';

    const params = new URLSearchParams(search);
    username = params.get('username');

    // After retrieveing the confirmation code from the user
    Auth.confirmSignUp(username, confirmationCode)
      /* eslint-disable no-unused-vars */
      .then(data => {
        this.setState({ confirmed: true });
      })
      .catch(err => alert(err.message));
  };

  render() {
    const { confirmationCode, confirmed } = this.state;

    if (confirmed) {
      return <Redirect to="/App" />;
    }

    return (
      <div id="loginScreenContainer">
        <div id="loginImageView">
          <img src={image} id="loginImage" alt="login" />
        </div>
        <div id="loginFormView">
          <div className="formViewBlock headerOrFooter" />
          <div className="formViewBlock">
            <div>
              <div>
                <img id="coreLogo" src={logo} alt="The CORE Logo" />
              </div>
              <p id="emailPromptText">Please check your email for a confirmation code.</p>
              <div>
                <FormField
                  onChange={this.onChangeCode}
                  header="Confirmation Code"
                  value={confirmationCode}
                  placeholder="000000"
                />
              </div>
              <div style={{ clear: 'both', paddingTop: 30 }}>
                <CustomButton onClick={this.onConfirm} text="Confirm Code" />
              </div>
            </div>
          </div>
          <div className="formViewBlock headerOrFooter" />
        </div>
      </div>
    );
  }
}

Confirm.propTypes = {
  location: PropTypes.object.isRequired,
};
