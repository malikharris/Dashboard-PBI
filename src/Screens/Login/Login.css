#loginScreenContainer {
  display: flex;
  flex: 1;
  height: 100%;
  flex-flow: row;
  /* IE stuff */
  -ms-flex-flow: row;
}

.formViewBlock {
  text-align: center;
  height: auto;
}

#loginImageView {
  width: 60%;
  height: 100%;
  overflow: hidden;
}

#loginImage {
  flex: 1;
}

#loginFormView {
  display: -webkit-flex;
  /* Safari */
  -webkit-align-items: center;
  /* Safari 7.0+ */
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  flex: 1;
}

#loginFormDiv {
  height: 15rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-bottom: 5rem;
}

#emailPromptText {
  /* Style for "Enter your" */
  width: 463px;
  height: 19px;
  color: #707070;
  font-family: Interstate;
  font-size: 16px;
  font-weight: 400;
  line-height: 13px;
}

#backButton {
  background-color: rgba(255, 252, 252, 0);
  border: solid 1px #008248;
  border-radius: 7px;
  margin-top: 20px;
  width: 100px;
}

#backButton:hover {
  transform: translateY(-7%);
  transition-duration: 100ms;
}

#coreLogo {
  width: 224px;
  height: 103px;
}

#signUpDiv {
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
}

#rememberMeSegment {
  flex: 1;
  float: left;
  font-family: Forza, sans-serif;
  color: #008348;
  font-weight: 500;
}

.rememberMeCheckbox {
  transform: scale(1.3);
  float: left;
}

#signUpSegment {
  background-color: #fff;
  cursor: pointer;
  flex: 1;
  float: right;
}

#signUpSegmentText {
  font-family: Forza, sans-serif;
  color: #008348;
  font-weight: 500;
  text-decoration: underline;
  font-size: 12pt;
}

#subcontractorLoginButtonContainer {
  padding-top: 25px;
  flex-direction: row;
  justify-content: center;
  clear: both;
}

#forgotPasswordView {
  background-color: #fff;
  border: none;
  cursor: pointer;
}

#forgotPasswordText {
  color: #008348;
  font-family: Forza, sans-serif;
  text-decoration: underline;
  font-size: 10pt;
}

#copyrightFooter {
  font-family: 'Interstate', sans-serif;
  line-height: 13px;
  font-weight: 400;
  font-size: 12px;
  color: #707070;
  /* margin-bottom: 68px; */
}

/* Set a max height to preserve the centering of the login form */
.headerOrFooter {
  height: 100px;
  width: 100%;
}
