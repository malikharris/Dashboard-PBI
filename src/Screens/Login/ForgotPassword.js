/* eslint-disable no-unused-vars */
import React from 'react';
import { Redirect } from 'react-router-dom';
import { Auth } from 'aws-amplify';
import image from '../../assets/images/Login_Image.svg';
import logo from '../../assets/images/CORE-logo.svg';

import FormField from '../../comps/FormField/FormField';
import CustomButton from '../../comps/CustomButton/CustomButton';
import { containsCOREDomain } from '../../utils';
import './Login.css';

export default class ForgotPassword extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      confirmationCode: '',
      email: '',
      newPassword: '',
      codeSent: false,
      confirmed: false,
    };
  }

  onChangeCode = e => {
    this.setState({ confirmationCode: e.target.value });
  };

  onChangeEmail = e => {
    this.setState({ email: e.target.value });
  };

  onChangeNewPassword = e => {
    this.setState({ newPassword: e.target.value });
  };

  onGetCode = () => {
    const { email } = this.state;

    if (!email.includes('@') || !email.includes('.')) {
      alert('Please enter a valid email.');
      return;
    }
    if (containsCOREDomain(email) && email !== 'admin@coreconstruction.com') {
      alert('Please use the CORE Employee Azure login portal to reset any passwords.');
      return;
    }

    // After retrieveing the confirmation code from the user
    Auth.forgotPassword(email)
      .then(data => {
        this.setState({ codeSent: true });
      })
      .catch(err => console.log(err));
  };

  onUpdatePassword = () => {
    const { email, confirmationCode, newPassword } = this.state;
    // Collect confirmation code and new password , then
    Auth.forgotPasswordSubmit(email, confirmationCode, newPassword)
      .then(data => {
        console.log(data);
        this.setState({ confirmed: true });
      })
      .catch(err => {
        alert(err.message);
        console.log(err);
      });
  };

  render() {
    const { confirmationCode, confirmed, codeSent, email, newPassword } = this.state;

    if (confirmed) {
      return <Redirect to="/Login" />;
    }

    return (
      <div id="loginScreenContainer">
        <div id="loginImageView">
          <img src={image} id="loginImage" alt="login" />
        </div>
        <div id="loginFormView">
          <div className="formViewBlock headerOrFooter" />
          <div className="formViewBlock">
            <div>
              <div>
                <img id="coreLogo" src={logo} alt="The CORE Logo" />
              </div>
              {codeSent === true ? (
                <div>
                  <p id="emailPromptText">Please enter the code and your new password.</p>
                  <div>
                    <FormField
                      onChange={this.onChangeCode}
                      header="Code"
                      value={confirmationCode}
                      placeholder="000000"
                    />
                    <FormField
                      onChange={this.onChangeNewPassword}
                      header="New Password"
                      value={newPassword}
                      placeholder="*************"
                    />
                  </div>
                  <div style={{ clear: 'both', paddingTop: 30 }}>
                    <CustomButton onClick={this.onUpdatePassword} text="Update Password" />
                  </div>
                </div>
              ) : (
                <div>
                  <p id="emailPromptText">
                    Please enter your email to receive a verification code.
                  </p>
                  <div>
                    <FormField
                      onChange={this.onChangeEmail}
                      header="Email"
                      value={email}
                      placeholder="janedoe@construction.com"
                    />
                  </div>
                  <div style={{ clear: 'both', paddingTop: 30 }}>
                    <CustomButton onClick={this.onGetCode} text="Get Code" />
                  </div>
                </div>
              )}
            </div>
          </div>
          <div className="formViewBlock headerOrFooter" />
        </div>
      </div>
    );
  }
}
