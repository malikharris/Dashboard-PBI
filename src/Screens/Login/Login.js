/* eslint-disable no-unused-vars */
import React, {useState, useEffect} from 'react';
import { Redirect } from 'react-router-dom';
import { Auth, Hub } from 'aws-amplify';
import { withOAuth } from 'aws-amplify-react';
import Background from '../../components/Background'
import Logo from '../../components/Logo';
import CustomButton from '../../components/CustomButton/CustomButton';
import FormField from '../../components/FormField/FormField';
import './Login.css';


const Login=(props)=>{
  const [state, setState] = useState({
    name: '',
    /* eslint-disable no-undef */
    email: localStorage.getItem('current_user_email') || '',
    password: '',
    confirmPassword: '',
    phoneNumber: '',
    user: undefined,
    showSignUp: false,
    rememberMe: !!localStorage.getItem('current_user_email'),
    forgotPassword: false,
  });

  const {name, email, password, confirmPassword, phoneNumber, user, showSignUp, rememberMe, forgotPassword} = state;

  useEffect(() => {
    Hub.listen('auth', data => {
      switch (data.payload.event) {
        case 'signIn':
          this.setState({ user: data.payload.data });
          break;
        case 'signIn_failure':
          console.log('login -> sign in error: ', JSON.stringify(data, null, 2));
          alert(
            `Error occured while attempting to sign in. Please share this error message: ${JSON.stringify(
              data,
              null,
              2,
            )}`,
          );
          break;
        default:
          break;
      }
    });

    Auth.currentAuthenticatedUser()
      .then(user => {
        this.setState({ user });
      })
      .catch(() => console.log('Not signed in'));
  }, []);

  const onLogin=()=>{
    console.log("Log in clicked")
  }
  const onChangeEmail = event => {
    setState({ email: event.target.value });
  };

  const onChangePassword = event => {
    setState({ password: event.target.value });
  };

  const onCheckRememberMe = e => {
    setState({ rememberMe: e.target.checked });
  };

  const onForgotPassword = () => {
    setState({ forgotPassword: true });
  };

    if (forgotPassword) {
      return <Redirect to="/ForgotPassword" />;
    }

    if (user && showSignUp) {
      return (
        <Redirect
          to={{
            pathname: '/Confirm',
            search: `?username=${email}`,
          }}
        />
      );
    }

    if (user) return <Redirect to="/App" />;

    return (

      <div id="loginScreenContainer">
      <Background width="60%"/>
        <div id="loginImageView"/>
        <div id="loginFormView">
          <div className="formViewBlock headerOrFooter" />
          <div className="formViewBlock">
            <div>
              <div>
                <Logo/>
              </div>
                <div>
                  <FormField
                    onChange={onChangeEmail}
                    header="Email"
                    value={email}
                    placeholder="johndoe@coreconstruction.com"
                  />
                  <div style={{ marginTop: 40 }}>
                    <FormField
                      onChange={onChangePassword}
                      header="Password"
                      value={password}
                      type="password"
                      placeholder="************"
                    />
                  </div>
                  <div id="signUpDiv">
                    <p id="rememberMeSegment">
                      <input
                        className="rememberMeCheckbox"
                        type="checkbox"
                        checked={rememberMe}
                        onChange={onCheckRememberMe}
                      />
                      <span style={{ marginTop: 3 }}> Remember Me</span>
                    </p>
                  </div>
                  <div id="subcontractorLoginButtonContainer">
                    <CustomButton onClick={onLogin} text="Login" />
                  </div>
                </div>
            </div>
          </div>
          <div className="formViewBlock headerOrFooter">
            <button type="button" id="forgotPasswordView" onClick={onForgotPassword}>
              <p id="forgotPasswordText">Forgot Password?</p>
            </button>
            <p id="copyrightFooter">
              {'Copyright '}
              {new Date().getFullYear()}
              {' The CORE Group LTD.'}
            </p>
          </div>
        </div>
      </div>
    );
  }




export default withOAuth(Login);
