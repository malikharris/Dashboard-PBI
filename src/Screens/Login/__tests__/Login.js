import React from 'react';
import { shallow } from 'enzyme';
import Login from '../Login';

it('should render Login correctly', () => {
  const wrapper = shallow(<Login />);
  expect(wrapper).toMatchSnapshot();
});

it('should reject invalid email input', () => {});

it('should accept valid email input', () => {});
