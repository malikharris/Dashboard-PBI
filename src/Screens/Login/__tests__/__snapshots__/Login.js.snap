// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login correctly 1`] = `
<Login
  OAuthSignIn={[Function]}
/>
`;
